Storage/container/one_to_one.h

```diff
@@ -42,65 +42,72 @@ template <typename T,
           typename T_DELETE_EVENT,
           template <typename...> class ROW_MAP,
           template <typename...> class COL_MAP>
-class GenericOne2One {
+class GenericOneToOne {
  public:
   using T_ROW = sfinae::ENTRY_ROW_TYPE<T>;
   using T_COL = sfinae::ENTRY_COL_TYPE<T>;
-  using T_RCPAIR = std::pair<T_ROW, T_COL>;
-  using T_WHOLE_MAP = std::unordered_map<T_RCPAIR, std::unique_ptr<T>, CurrentHashFunction<T_RCPAIR>>;
+  using T_KEY = std::pair<T_ROW, T_COL>;
+  using T_ELEMENTS_MAP = std::unordered_map<T_KEY, std::unique_ptr<T>, CurrentHashFunction<T_KEY>>;
   using T_FORWARD_MAP = ROW_MAP<T_ROW, const T*>;
   using T_TRANSPOSED_MAP = COL_MAP<T_COL, const T*>;
   using T_REST_BEHAVIOR = rest::behavior::Matrix;
 
-  explicit GenericOne2One(MutationJournal& journal) : journal_(journal) {}
+  explicit GenericOneToOne(MutationJournal& journal) : journal_(journal) {}
 
   bool Empty() const { return map_.empty(); }
   size_t Size() const { return map_.size(); }
 
+  // Adds specified object, removes and/or overwrites all existing entries
+  // with row and/or col value matching the corresponding values of the new entry
   void Add(const T& object) {
     const auto row = sfinae::GetRow(object);
     const auto col = sfinae::GetCol(object);
-    const auto row_col = std::make_pair(row, col);
-    const auto it = map_.find(row_col);
+    const auto key = std::make_pair(row, col);
+    const auto it = map_.find(key);
     if (it != map_.end()) {
-      const T prev_object = *(it->second);
+      const T previous_object = *(it->second);
       journal_.LogMutation(T_UPDATE_EVENT(object),
-                           [this, row_col, prev_object]() { DoAdd(row_col, prev_object); });
+                           [this, key, previous_object]() { DoAdd(key, previous_object); });
     } else {
       const auto it_row = forward_.find(row);
       const auto it_col = transposed_.find(col);
-      bool row_occupied = it_row != forward_.end();
-      bool col_occupied = it_col != transposed_.end();
+      const bool row_occupied = (it_row != forward_.end());
+      const bool col_occupied = (it_col != transposed_.end());
       if (row_occupied && col_occupied) {
-        const T prev_object1 = *(it_row->second);
-        const T prev_object2 = *(it_col->second);
-        const auto row_col1 = std::make_pair(row, sfinae::GetCol(prev_object1));
-        const auto row_col2 = std::make_pair(sfinae::GetRow(prev_object2), col);
-        journal_.LogMutation(T_DELETE_EVENT(prev_object1),
-                             [this, row_col1, prev_object1]() { DoAdd(row_col1, prev_object1); });
-        journal_.LogMutation(T_DELETE_EVENT(prev_object2),
-                             [this, row_col2, prev_object2]() { DoAdd(row_col2, prev_object2); });
-        DoErase(row_col1);
-        DoErase(row_col2);
+        const T previous_object_same_row = *(it_row->second);
```

Comment:
Why aren't these const references? They are captured by value in the lambda regardless.